microprofile/metrics-smallrye/src/main/java/org/wildfly/extension/microprofile/metrics/MicroProfileMetricsSubsystemDefinition.java

```diff
@@ -90,4 +102,46 @@ protected MicroProfileMetricsSubsystemDefinition() {
     public Collection<AttributeDefinition> getAttributes() {
         return Arrays.asList(ATTRIBUTES);
     }
+
+    @Override
+    public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
+        final AttributeDefinition[] attrs = {SECURITY_ENABLED, PREFIX};
+        ReloadRequiredWriteAttributeHandler handler = new ReloadRequiredWriteAttributeHandler(attrs);
+        resourceRegistration.registerReadWriteAttribute(SECURITY_ENABLED, null, handler);
+        resourceRegistration.registerReadWriteAttribute(EXPOSED_SUBSYSTEMS, null, new ExposedSubsystemWriterHandler());
+        resourceRegistration.registerReadWriteAttribute(PREFIX, null, handler);
+    }
+
+    private class ExposedSubsystemWriterHandler extends ReloadRequiredWriteAttributeHandler {
+        private ExposedSubsystemWriterHandler() {
+            super(EXPOSED_SUBSYSTEMS);
+        }
+
+        @Override
+        protected void finishModelStage(final OperationContext context, final ModelNode operation, String attributeName,
+                                        ModelNode newValue, ModelNode oldValue, final Resource model) throws OperationFailedException {
+            if (newValue.isDefined()) {
+                Set<String> subsystems = context.readResourceFromRoot(EMPTY_ADDRESS).getChildrenNames("subsystem");
```

Comment:
It would be better to just read the subsystem children and avoid read the model recursively. 

I suggest using `context.readResourceFromRoot(PathAddress.pathAddress(PathElement.pathElement(SUBSYSTEM)), false).getChildrenNames(SUBSYSTEM);` instead